File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdResetEvent.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdResetEvent_call(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong event, jlong stageMask){
    (void)env;
    (void)jc;
    PFN_vkCmdResetEvent f = (PFN_vkCmdResetEvent)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkEvent*)jniLongToPointer(event)),
        *((VkPipelineStageFlags*)jniLongToPointer(stageMask))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkCmdResetEvent.java
package cz.mg.vulkan;

public class PFNvkCmdResetEvent extends VkFunctionPointer {
    public PFNvkCmdResetEvent() {
    }

    protected PFNvkCmdResetEvent(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdResetEvent(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdResetEvent(long value) {
        setValue(value);
    }

    public PFNvkCmdResetEvent(VkInstance instance) {
        super(instance, new VkString("vkCmdResetEvent"));
    }

    public void call(VkCommandBuffer commandBuffer, VkEvent event, int stageMask){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), event != null ? event.getVkAddress() : VkPointer.getNullAddressNative(), stageMask);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, long event, int stageMask);

}
